### DISCUSSION TITLE
Content References

### DISCUSSION THREAD

**Discussion by bjarnef** [umbraco/Umbraco-CMS#11588] [Features and ideas]
Currently Media and Data Types have references to where these entities are used.

However Content doesn't have this, yet.

It would be great to have this as well for Content. From time to rime, when working with Umbraco Cloud and Deploy, we are getting dependency errors on restore, because a reference node has been deleted.

For example an employee node picked from MTNP or just a link for link picker in Rich Text editor which had a node selected, which afterwards has been deleted.

So I would be really useful on a Content node to see if other nodes have dependency on this before deleting the node itself 👍
 
Maybe there should even be a warning to content editors if trying to delete a node, which are referenced to from other nodes?

**Comment by dawoe** [comment]
Hi @bjarnef 

Umbraco actually tracks the references... but does not show them in the UI. I created some PR's during Hacktoberfest 2020 to improve this but they still haven't been merged yet. See the related issue : https://github.com/umbraco/Umbraco-CMS/issues/8679

Because of lack of time and response from HQ I stopped working on them. You could use my Nexu package which has all this functionality for content and media.

Dave

**Reply by bjarnef** [reply]
Ahh cool. I saw some work has been done, but just noticed it isn't shown in the UI yet.
I would definitely be a big improvement and help a lot on Umbraco Cloud projects.

Furthermore I have often noticed content editors have selected nodes on linked to content nodes via link picker in richtext editor - either actual content or content templates. If/when these content nodes later are deleted (either moved to recycle bin or deleted completely) it cause some dependency errors when restoring content from this environment.

At the moment I don't think there are any warning in the delete dialog if content editors delete a node, which is a dependency on antoher node.

@nul800sebastiaan @AndyButland not sure if this in on the roadmap, but it would be a really great feature 😎

**Reply by AndyButland** [reply]
Tangentially related, but just wanted to mention there's a feature in the latest Deploy RC which can help with this issue with restoring when related content or media is in the recycle bin or deleted.  See [this config](https://our.umbraco.com/documentation/Add-ons/Umbraco-Deploy/Deploy-Settings/#ignorebrokendependenciesbehavior) - allowing you to ignore broken dependencies just for certain operations (e.g. restore from upstream, when you are probably OK with it, but not transfer to upstream, when you probably are not).

**Reply by ravimotha** [reply]
Following up on @bjarnef initials comment 
_For example an employee node picked from MTNP or just a link for link picker in Rich Text editor which had a node selected, which afterwards has been deleted.

So I would be really useful on a Content node to see if other nodes have dependency on this before deleting the node itself 👍

Maybe there should even be a warning to content editors if trying to delete a node, which are referenced to from other nodes?_

If this becomes  a reality i think this and having something like the feature @dawoe mentioned exposed would greatly enhance the editor user experience
